packages/playwright-core/src/server/android/android.ts

```diff
@@ -256,15 +260,23 @@ export class AndroidDevice extends SdkObject {
     await androidBrowser._init();
     this._browserConnections.add(androidBrowser);
 
+    const artifactsDir = await fs.promises.mkdtemp(ARTIFACTS_FOLDER);
```

Comment:
we don't launch a process, so we have to implement a naive cleanup logic ourselves.